WalletWasabi.Fluent/Helpers/StartupHelper.cs

```diff
@@ -0,0 +1,61 @@
+using Microsoft.Win32;
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using WalletWasabi.Logging;
+
+namespace WalletWasabi.Fluent.Helpers
+{
+	public static class StartupHelper
+	{
+		public const string StartupErrorMessage = "Something went wrong while trying to make your changes.";
+		private const string KeyPath = "SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Run";
+
+		public static bool TryModifyStartupSetting(bool runOnSystemStartup)
+		{
+			if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+			{
+				string pathToExeFile = Assembly.GetExecutingAssembly().Location;
+				pathToExeFile = pathToExeFile.Remove(pathToExeFile.Length - 4);        // This part has to change if this gets released
+				pathToExeFile += ".Desktop.exe";
```

Comment:
The .exe file have to exist, because that's our entry point for the app. With your earlier suggestion, we don't even have to remove or add anything to the `pathToExeFile`, so I think this is done.

Reply:
`Assembly.GetEntryAssembly().Location;` gives you a DLL file and not an exe file. Apparently, there is some history to it because in .NET Framework it used to return `.exe` but in .NET (Core) it returns `.dll` files. So I think it will be necessary to do: `string pathToExeFile = Assembly.GetEntryAssembly().Location[..^4] + ".exe"`

Adding a check to verify that the path really exists is not required but nice to have. Possible silent errors are more time costly than writing an if statement now even if it is never hit.

Reply:
Oh, you are right again. I will add a check for it too.